processor/ratelimitprocessor/local.go

```diff
@@ -56,9 +56,17 @@ func (r *localRateLimiter) RateLimit(ctx context.Context, hits int) error {
 	key := getUniqueKey(ctx, r.cfg.MetadataKeys)
 	v, _ := r.limiters.LoadOrStore(key, rate.NewLimiter(rate.Limit(r.cfg.Rate), r.cfg.Burst))
 	limiter := v.(*rate.Limiter)
-	if ok := limiter.AllowN(time.Now(), hits); !ok {
-		// TODO add configurable behaviour for returning an error vs. delaying processing.
-		return errTooManyRequests
+	switch r.cfg.OnThrottle {
+	case OnThrottleError:
+		if ok := limiter.AllowN(time.Now(), hits); !ok {
+			return errTooManyRequests
+		}
+	case OnThrottleBlock:
+		r := limiter.ReserveN(time.Now(), hits)
+		if !r.OK() {
+			return errTooManyRequests
+		}
+		time.Sleep(r.Delay())
```

Comment:
Thanks for this change, the non stop errors make it really hard to read anything else.

So just to be clear: if we have the pipeline loadgen -> ratelimit -> debug, once ratelimit sleeps, the whole process gets blocked?

Reply:
> So just to be clear: if we have the pipeline loadgen -> ratelimit -> debug, once ratelimit sleeps, the whole process gets blocked?

Correct

Reply:
I've tested it with https://github.com/elastic/opentelemetry-collector-components/pull/246 and there are no more "too many requests" flooding in logs